lib/wasix/src/runtime/resolver/wapm_source.rs

```diff
@@ -86,9 +86,9 @@ impl Source for WapmSource {
 
         for pkg_version in versions {
             tracing::trace!(?pkg_version, "checking package version");
-
             let version = Version::parse(&pkg_version.version)?;
-            if version_constraint.matches(&version) {
+
+            if !pkg_version.is_archived && version_constraint.matches(&version) {
```

Comment:
so archived version can never be selected, even with the precise version match? Probably ok for most cases anyway.